geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientUpdater.java

```diff
@@ -1763,6 +1767,19 @@ private void processMessages() {
     }
   }
 
+  private void handleAuthenticate(Message clientMessage) {
+    // if client is in multi-user mode, the CacheClientUpdator (at this point)
+    // can't differentiate which user this message is intended to. so throw exception for now
+    // one possible solution is re-authenticate all users in this client
+    if (qManager.getPool().getMultiuserAuthentication()) {
+      throw new UnsupportedOperationException(
```

Comment:
The test shows the connection gets closed, may need to investigate what eventually triggered the close event.

Reply:
This is getting processed by CacheClientUpdater thread...As you can see from the thread name, this thread is used only for local cache operation; its not used to do client-to-server operation. I am worried about the exception it could see by doing server side operation....And its impact on overall closing CCU thread and re-creating it. In the past we had several issues with older and new CCU threads running simultaneously. Without sufficient tests its hard to see its impact. It will be nice to process this in separate thread than the CCU thread.

Reply:
I didn't put this "re-authenticate" in my initial implementation because
1) if we get this message from the server, no other message will be delivered to this client until we re-authenticate back successfully.
2) even if we put it in a separate thread, we will need to see if any exception happens on that thread and if so, we need to close the thread anyway.

I haven't seen much benefit of putting this in a different thread yet, but I don't see much harm in putting in a different thread either. Since it can go either way, I much rather wait for this PR to settle before I put in more "improvement".

Reply:
@jinmeiliao my concerns are:
Point 1:
Yes that is true.
Point 2:
If there is exception, then thread can handle according to the exception. And if it needs to close the connection, it can. But what I am trying to convey here is the CacheClientUpdater thread is not meant to be used for server cache operation. And currently its handling the exception based on the expectation of server to client communication. I am not seeing any tests/code changes in CacheClientUpdater exception handling to handle the exception from client to server communication which this PR is introducing.